finality-aleph/src/sync/handler/mod.rs

```diff
@@ -2879,4 +2895,53 @@ mod tests {
             }
         }
     }
+
+    #[tokio::test]
+    async fn accepts_chain_extension_branch() {
+        let (mut h1, mut b1, mut n1, genesis) = setup();
+        let (mut h2, _b2, _n2, _genesis) = setup();
+
+        // Make h1 aware of 2 headers.
+        let branch = grow_light_branch(&mut h1, &genesis, 2, 0);
+
+        // Import the corresponding 2 blocks in h1.
+        for h in branch.clone() {
+            let block = MockBlock::new(h.clone(), true);
+            b1.import_block(block, false);
+            match n1.next().await {
+                Ok(BlockImported(header)) => {
+                    h1.block_imported(header)
+                        .expect("block imported should succeed");
+                }
+                _ => panic!("should notify about imported block"),
+            }
+        }
+
+        println!("kurcze 1");
```

Comment:
Little chickens are unfortunately not welcome in our code, as that would make it non-vegan.

Reply:
Apologies, no more chickens :pray: